File: backend/components/metiers/threed/legacy/migration/models/migration_model.js
// Modèle de données de migration legacy Threed (JS)
// Ultra avancé, clé en main, conforme au cahier des charges
class MigrationLog {
  constructor(source, target, status, details = {}) {
    this.source = source;
    this.target = target;
    this.status = status;
    this.details = details;
  }
  toJSON() {
    return {
      source: this.source,
      target: this.target,
      status: this.status,
      details: this.details,
    };
  }
}
module.exports = { MigrationLog };

File: backend/components/metiers/threed/legacy/migration/models/migration_model.py
"""
Modèle de données de migration legacy Threed (Python)
Ultra avancé, clé en main, conforme au cahier des charges.
"""


class MigrationLog:
    def __init__(self, source, target, status, details=None):
        self.source = source
        self.target = target
        self.status = status
        self.details = details or {}

    def to_dict(self):
        return {
            "source": self.source,
            "target": self.target,
            "status": self.status,
            "details": self.details,
        }
